src/test/java/com/notably/storage/JsonBlockStorageTest.java

```diff
@@ -0,0 +1,109 @@
+package com.notably.storage;
+
+import static com.notably.testutil.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import com.notably.commons.exceptions.DataConversionException;
+import com.notably.commons.path.AbsolutePath;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeImpl;
+import com.notably.testutil.TypicalBlockTree;
+
+public class JsonBlockStorageTest {
+    private static final Path TEST_DATA_FOLDER = Paths.get("src", "test", "data", "JsonBlockStorageTest");
+
+    @TempDir
+    public Path testFolder;
+
+    @Test
+    public void readBlockTree_nullFilePath_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> readBlockTree(null));
+    }
+
+    private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
+        return prefsFileInTestDataFolder != null
+                ? TEST_DATA_FOLDER.resolve(prefsFileInTestDataFolder)
+                : null;
+    }
+
+    private java.util.Optional<BlockTree> readBlockTree(String filePath) throws Exception {
+        return new JsonBlockStorage(Paths.get(filePath)).readBlockTree(addToTestDataPathIfNotNull(filePath));
+    }
+
+    /**
+     * Saves the {@code BlockTree} at the specified path.
+     *
+     * @param blockTree
+     * @param filePath
+     */
+    private void saveBlockTree(BlockTree blockTree, String filePath) {
+        try {
+            new JsonBlockStorage(Paths.get(filePath))
+                    .saveBlockTree(blockTree, addToTestDataPathIfNotNull(filePath));
```

Comment:
Not sure if I'm getting this correctly, but I believe `new JsonBlockStorage(Paths.get(filePath)).saveBlockTree(blockTree);` would suffice?

Reply:
`addToTestDataPathIfNotNull(filePath)` creates a `Path` object from the `filePath` which is actually a String, so this is needed!